third-party/csp-evaluator/optimized_binary.js

```diff
@@ -0,0 +1,160 @@
+b.push(a)}return b}goog.exportSymbol("evaluateForFailure",module$contents$google3$javascript$security$csp$csp_evaluator$lighthouse$lighthouse_checks_evaluateForFailure);goog.exportSymbol("evaluateForWarnings",module$contents$google3$javascript$security$csp$csp_evaluator$lighthouse$lighthouse_checks_evaluateForWarnings);goog.exportSymbol("evaluateForSyntaxErrors",module$contents$google3$javascript$security$csp$csp_evaluator$lighthouse$lighthouse_checks_evaluateForSyntaxErrors);var module$exports$google3$javascript$security$csp$csp_evaluator$parser={CspParser:class{constructor(a){this.csp=new module$exports$google3$javascript$security$csp$csp_evaluator$csp.Csp;this.parse(a)}parse(a){this.csp=new module$exports$google3$javascript$security$csp$csp_evaluator$csp.Csp;a=a.split(";");for(let d=0;d<a.length;d++){var b=a[d].trim();b=b.match(/\S+/g);if(Array.isArray(b)){const c=b[0].toLowerCase();if(c in this.csp)continue;(0,module$exports$google3$javascript$security$csp$csp_evaluator$csp.isDirective)(c);
+const e=[];for(let f,g=1;f=b[g];g++)f=module$contents$google3$javascript$security$csp$csp_evaluator$parser_normalizeDirectiveValue(f),e.includes(f)||e.push(f);this.csp[c]=e}}return this.csp}}};
+function module$contents$google3$javascript$security$csp$csp_evaluator$parser_normalizeDirectiveValue(a){a=a.trim();const b=a.toLowerCase();return(0,module$exports$google3$javascript$security$csp$csp_evaluator$csp.isKeyword)(b)||(0,module$exports$google3$javascript$security$csp$csp_evaluator$csp.isUrlScheme)(a)?b:a}module$exports$google3$javascript$security$csp$csp_evaluator$parser.TEST_ONLY={normalizeDirectiveValue:module$contents$google3$javascript$security$csp$csp_evaluator$parser_normalizeDirectiveValue};
+goog.exportSymbol("CspParser",module$exports$google3$javascript$security$csp$csp_evaluator$parser.CspParser);
+
+module.exports = {
+  Parser: module$exports$google3$javascript$security$csp$csp_evaluator$parser.CspParser,
+  Type: module$exports$google3$javascript$security$csp$csp_evaluator$finding.Type,
+  Directive: module$exports$google3$javascript$security$csp$csp_evaluator$csp.Directive,
+  evaluateForFailure: module$contents$google3$javascript$security$csp$csp_evaluator$lighthouse$lighthouse_checks_evaluateForFailure,
+  evaluateForWarnings: module$contents$google3$javascript$security$csp$csp_evaluator$lighthouse$lighthouse_checks_evaluateForWarnings,
+  evaluateForSyntaxErrors: module$contents$google3$javascript$security$csp$csp_evaluator$lighthouse$lighthouse_checks_evaluateForSyntaxErrors,
+}
```

Comment:
This is not in the original binary.

Reply:
I suppose part of publishing to npm will be to add this export declaration?

Reply:
Good time to pull in @lweichselbaum and @salcho.

Is it possible to add this or some similar export in the npm module?

Reply:
Hey folks, thanks for the mention, a couple comments:

We are evaluating what is needed to build an NPM artefact from the CSP evaluator. The TL;DR is that we could provide a quick, temporary solution to unblock LH by, say, publishing the raw CSP evaluator binary to GitHub, etc. while we make the necessary changes to build a proper NPM artefact. More details on this next week!

All the exports listed here LGTM - Just to make sure we're on the same page:

- `Parser` & `Directive` is what LH uses to parse the CSP
- `evaluateFor*` is what LH uses to get errors/warnings from CSP
- `Type` is used to know which `Finding`s belong to which checks - shouldn't `Finding` itself and `Severity` be exported as well?

Reply:
> The TL;DR is that we could provide a quick, temporary solution to unblock LH by, say, publishing the raw CSP evaluator binary to GitHub, etc.

FWIW I don't see any reason we couldn't land this with the current binary. An npm module would just make the process of importing and updating CSP evaluator much easier.

> `Parser` & `Directive` is what LH uses to parse the CSP

We use `Directive` to match findings to translated UI strings.

> `evaluateFor*` is what LH uses to get errors/warnings from CSP

Correct.

> shouldn't `Finding` itself and `Severity` be exported as well?

We aren't currently using `Severity` for anything, but I could see us using it for severity icons in the future. Exporting that as well sounds good to me.

`Finding` doesn't need to be exported because we only need the finding type information which we redefine in `csp-evaluator.js`:
https://github.com/GoogleChrome/lighthouse/blob/ee8624b97f864ac900483adc80dec428b5edd76c/lighthouse-core/lib/csp-evaluator.js#L8-L15

Reply:
have y'all investigated turning on advanced/recommended optimizations? Everything's very flat here but those long names could probably lose a lot more bytes :)

Reply:
Thanks for the clarifications! It seems like we're good to go to land this PR with the current binary and we have the minimal set of exports needed for the integration. LGTM!